### ISSUE TITLE
Bug: Stuck at Taking you to your organization's sign-in page for microsoft services

### ISSUE THREAD

**Issue by jbreitbart** [ferdium/ferdium-app#443]
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

6.0.0-nightly.91

### What Operating System are you using?

Ubuntu

### Operating System Version

20.04

### What arch are you using?

x64

### Expected Behavior

When I try to login with my work email I get redirected to the signup page of my company at which I can authorize myself.

### Actual Behavior

The redirect is not working. I am stuck at a page that says Taking you to your organization's sign-in page.

### Steps to reproduce

1. Add Microsoft teams
2. Hit the sign-in options
3. Choose sign in to an organization
4. Enter a valid organization name
5. Enter a valid email
6. I am stuck a redirect page

### Screenshots

![image](https://user-images.githubusercontent.com/881727/177784195-1dfb3218-4689-496f-b66d-928fa28b4e3a.png)
I am stuck at this page

**Comment by vraravam** [comment]
I also use Ferdium with hosted Teams and Outlook for my client. I dont see this "stuck" issue. Maybe your session is somehow corrupt. Can you please delete the service and re-add it and try once more?

**Comment by jbreitbart** [comment]
I have removed / re-added the service multiple times. I have also tried adding Outlook, but that shows the same behavior. Is there a cache/config directory that I should remove to be sure? I don't mind loosing my current configuration.

**Comment by vraravam** [comment]
Do these questions/answers help: https://ferdium.org/faq ?
If not, you can go to the Preferences page inside Ferdium, go into `Advanced`, scroll down, and you will find some buttons to open your Ferdium profile folder. Clicking on this should open your file manager application to the appropriate directory. Close Ferdium, and you can delete the `Partitions` and `Session Storage` folder. Please remember that your logged-in sessions for all services will be lost and you will have to log back into each one of them by doing so.

**Comment by jbreitbart** [comment]
Thanks for the link to the FAQ, but that did not help me solve the issue. I have removed the `Partition` and `Sessation Storage`, but this did not change the behavior I am seeing. The login page is still not opening.

**Comment by vraravam** [comment]
IS there any error in the `View > Toggle Service Developer tools`? (There are 2 dev tools - one for the main app, and one for the currently active service. please check the service-specific one and add a screenshot here?
Also, what version of Ferdium are you running and what version of the service recipe itself?

**Comment by vraravam** [comment]
I'm on the latest Ferdium (nightly v94), and MS Teams service recipe is at 3.2.1)
<img width="1026" alt="Screenshot 2022-07-11 at 3 00 32 PM" src="https://user-images.githubusercontent.com/69629/178234644-201856f7-0191-4881-ab5b-390782f3d6d2.png">

**Comment by MikeDabrowski** [comment]
I got similar problem. MS says I need to install the extension to access organisation specific pages (teams).

> Since you're using Chrome, you need to install this [extension](https://chrome.google.com/webstore/detail/windows-10-accounts/ppnbnpeolgkicgegkbkbjmhlideopiji). You must be on Windows 10 version 1703 and above. Alternatively, you can use Microsoft Edge or Internet Explorer to access this application.

![image](https://user-images.githubusercontent.com/17836911/178425233-328389f6-5221-4162-8705-99ef8aafc28b.png)

I will try updating ferdium for now, but I doubt it will help

**Comment by vraravam** [comment]
Can you please try to change the user-agent string for your service - to one that mimics macos? I dont have this problem at all!

**Comment by MikeDabrowski** [comment]
@vraravam

i used

`Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36`
 
and it worked ! Pretty neat. However in the future I think we will need extension sideload, I've seen there is already feature request for that. I just tried to get access to app/session from user.js and webview.js but sadly it is not exposed there

**Comment by vraravam** [comment]
Closing since a `user-agent`-based logic is not something that we can control. This was the reason for this feature to be exposed to the ui so that user can choose what works for them

**Comment by MikeDabrowski** [comment]
I disagree, the solution you provided seems like a workaround. This time it worked but the next time it may not. The official way for teams is to use their extension. I'm not even sure if the OP's @jbreitbart problem can be solved by modifying the UA. How did you even know that this may be UA ?

**Comment by vraravam** [comment]
@MikeDabrowski - we can reopen this ticket - but, if the user-agent is the only differentiating factor (as per your comment up above), then there's nothing that we can do to change the behavior on Ferdium's end.
Please remember that all that Franz/Ferdi/Ferdium and its ilk are doing are rendering web-pages inside chromium tabs. Any deduction by the server to then render mobile vs desktop or turning on/off of features are somethings that's being done on the server-side and **_we have zero control on that behavior_** (other than changing the UA). 

Your next question is also equally valid: "what if this UA stops working in the future?" That has also happened with google services in the past. If so, then the user community rallies around to figure out which UA will work. This has repeated quite a few times in the past when I was contributing to Ferdi - which is what prompted me to implement the feature to expose this in the UI in the first place.

Do you still want to reopen? I would be glad to do so - but, we wont work on this unless there's a clear explanation of something that Ferdium is processing incorrectly.

**Comment by MikeDabrowski** [comment]
ok I understand your POV. Yet this is still a workaround that just happens to work in this case. I mean I doubt that giant companies will let it last for long. So lets leave this closed and revisit if the issue comes back. 

Meanwhile, the official way - installing the extension - is something already tracked [here](https://github.com/ferdium/ferdium-app/issues/135) and I believe worth adding since it not only may fix this issue but also other services (if they decide to create their own extension driven security models) and finally add more functionalities like password managers.

**Comment by vraravam** [comment]
yes - extension-support is in our radar, we will need to fix 1 more major upgrade of an internal dependency - and then will start on this aspect.

**Comment by jbreitbart** [comment]
So for me changing the user agent does not change anything. I also do not get a prompt asking me to install an extension. I get this prompt after login, however I am stuck before that point.

I did update my installation to 

Version: 6.0.0-nightly.98
Server: No Server
Electron: 19.0.8
Chrome: 102.0.5005.148
Node.js: 16.14.2
Platform: Linux 5.13
Arch: x64
Build date: Tue Jul 12 2022 04:46:35 GMT+0200 (Central European Summer Time)
Git SHA: 8bd534e6
Git branch: nightly

but this did not change anything for me. I have the following errors in the Developer Tools console

![image](https://user-images.githubusercontent.com/881727/178488837-acaead06-ba48-4672-8fbe-ab42f940e4ac.png)

**Comment by MikeDabrowski** [comment]
@jbreitbart 
> I have the following errors in the Developer Tools console
> 
> ![image](https://user-images.githubusercontent.com/881727/178488837-acaead06-ba48-4672-8fbe-ab42f940e4ac.png)

I think this is Ferdium Dev Tools, 
There is also Service Dev Tools:  
![image](https://user-images.githubusercontent.com/17836911/178493330-1df04943-88fa-43b7-84ca-b4fe38ea28b1.png)
 which will show the errors originating from the teams app. Can you paste the screenshot of it ?

**Comment by jbreitbart** [comment]
I just tried it again, but I this is the output I get from the Service Developers Tools

![image](https://user-images.githubusercontent.com/881727/178529492-58578861-8af8-4242-9f0f-7fda2e1d87aa.png)

However, the messages are already there before I enter my email address. The other dev tools show me

![image](https://user-images.githubusercontent.com/881727/178530581-bb291e5b-945e-4b0a-9d3e-72f977a23987.png)

when I enter my email adress

**Comment by SlanDF** [comment]
@jbreitbart & @MikeDabrowski could you create a new office service, select selfhosted and put `https://outlook.office365.com/mail/`  and try to log in?

**Comment by vraravam** [comment]
@SlanDF - I have used the same default service, but opted for the "Self-hosted" version - and used `https://outlook.office365.com/mail/` which works perfectly for me.

**Comment by MikeDabrowski** [comment]
@jbreitbart I also have the same errors in the log but still am able to log in and use teams.

**Comment by SlanDF** [comment]
> @jbreitbart I also have the same errors in the log but still am able to log in and use teams.

The console errors are related to darkmode, shouldnt affect your session

**Comment by MikeDabrowski** [comment]
Yes I can see that. Unfortunately I cannot offer any more input on that

**Comment by SlanDF** [comment]
@jbreitbart  could you try this?
1. Create a new office 365 service with self-hosted selected and paste `https://outlook.office365.com/mail/` in it
![image](https://user-images.githubusercontent.com/13062045/178730123-c7d63dd0-3d4d-4986-b4a2-7414a2e392e4.png)

2. Go to the newly created service
3. Press CTRL+B to open the navigation bar
4. Check if before typing in your email, the URL on the navigation bar is `login.microsoftonline.com`
5. Type your email and click next
6. Check if that URL in the navigation bar changed and see if you can login now

**Comment by MikeDabrowski** [comment]
Not me, I got the teams working. @jbreitbart is the OP here :D

**Comment by SlanDF** [comment]
> Not me, I got the teams working. @jbreitbart is the OP here :D

changed my comment, thought you were still stuck with the same issue of office

**Comment by jbreitbart** [comment]
@SlanDF thanks for the suggestion

>     4. Check if before typing in your email, the URL on the navigation bar is `login.microsoftonline.com`

The domain is correct, but there are various additional parameters in the url.

>     5. Type your email and click next
> 
>     6. Check if that URL in the navigation bar changed and see if you can login now

It is not changed. In the dev tools I can see the following

![image](https://user-images.githubusercontent.com/881727/178731510-9c201922-9147-4e2e-be87-acaef5370a26.png)

The URL it tries to access requires me to be connected via VPN (which I am) and to use a proxy. Which I configured in the service

![image](https://user-images.githubusercontent.com/881727/178731908-f67bcf4c-7d84-4789-b214-1a85ba2e1906.png)

If I do not configure the proxy the behavior is the same.

**Comment by SlanDF** [comment]
@jbreitbart you available to go for a quick call and screen share? Have some things I want to check from your end.
If you're on the Ferdium discord channel just send me a dm there

**Comment by M0UL** [comment]
i have the same problem with office 365 outlock.  with every reload it  loads the login page again.

**Comment by theretohere** [comment]
Outlook 365 in Ferdium gives me the login page on every reload also. 

But when I enter the custom URL https://outlook.office365.com/mail/ , after a reload the screen reloads about 10 times (url's cycle in the window title bar), then I get a 500 Something went wrong error, Then i press refresh the page, and I get dropped into my already logged in Outlook.

Ferdium 6.0.0 stable on Linux Mint 20

**Comment by jbreitbart** [comment]
> @jbreitbart you available to go for a quick call and screen share? Have some things I want to check from your end. If you're on the Ferdium discord channel just send me a dm there

Sorry for the delayed response. I am afraid I cannot use discord on that machine and chat / screen sharing in general is quite limited due to corporate policies.

**Comment by danyalaytekin** [comment]
Similar issue here with macOS and Outlook for Web, but in my case I have:

1. One Office 365 service that has stayed signed in for a couple of weeks now.
2. A second Office 365 service that shows the login page whenever Ferdium is restarted.  From there, it only needs the email address to be provided to log in (no password or 2FA required).

---

Update: seem to have fixed it for now by using self-hosted and https://outlook.office.com/mail/ for (2).  Solution seems a bit sus though.  (1) has not been touched and continues to arrive at https://outlook.office365.com/mail/.
